### PR TITLE
Fix Annulus width check

### PR DESCRIPTION
**Pull request by drbergman**
Fix check on annulus width to reflect the error message printed. The logic is presented in this (slightly) convoluted form to better conform with the error message. Obviously, the more concise `if min(self.a, self.b) < width:` works just as well.

This fix will allow annuli to be equivalent to Circles (in terms of area plotted, not underlying object structure). This means plotting (`_recompute_path`, I believe) could be improved by checking for `if w==a or w==b: PLOT WITHOUT HOLE`. Not sure how that would work, but everything does seem to work even without that change.

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: lib/matplotlib/patches.py [hunk 1]
```diff
@@ -1840,7 +1840,7 @@ def set_width(self, width):
         ----------
         width : float
         """
-        if min(self.a, self.b) <= width:
+        if width <= min(self.a, self.b) is False:
```
Comment (QuLogic) [inline lib/matplotlib/patches.py:1843]: The comparison plus `is False` is a bit odd; either use `not()` or invert the condition.

File: lib/matplotlib/patches.py [hunk 2]
```diff
@@ -1840,7 +1840,7 @@ def set_width(self, width):
...
             raise ValueError(
                 'Width of annulus must be less than or equal semi-minor axis')
```
Comment (QuLogic) [inline lib/matplotlib/patches.py:1845]: While you're here, could you also fix the typo? There should be a 'to' between equal and semi-minor.
```suggestion
                'Width of annulus must be less than or equal to semi-minor axis')
```

File: lib/matplotlib/patches.py (hunk 1 above)
Comment (timhoffm) [reply lib/matplotlib/patches.py:1843]:
```suggestion
        if width > min(self.a, self.b):
```

**Review by QuLogic — APPROVED** [review]
Probably can squash when merging.

**Review by timhoffm — APPROVED** [review]

**Comment by timhoffm** [comment]
Thanks and congratulations on your first contribution to Matplotlib 🚀 . We hope to see you again.

**Comment by drbergman** [comment]
Wow! < 24 h to get this in! Thank you so much. Glad to contribute.

**Pull request merged**